### DISCUSSION TITLE
DWD Warnung in Belchertown skin wie auf https://www.woellsdorf-wetter.de

### DISCUSSION THREAD

**Discussion by teffi71** [roe-dl/weewx-DWD#71] [Ideas]
Magst Du eine Anleitung schreiben, was Du verändert hast, damit die DWD Wetterwarnung verwendet wird?
<img width="745" alt="Screenshot 2024-08-12 at 12 41 22" src="https://github.com/user-attachments/assets/9ba1e844-91b7-4c44-9f13-4741616b7497">

Z.Z. wird bei mir gar keine Wetterwarnung angezeigt, obwohl weewx-DWD eine Warnung heruntergeladen hat.

**Comment by roe-dl** [comment]
Man muß die Warnung schon irgendwo einfügen, damit sie angezeigt wird. Ein Beispiel ist [hier](https://github.com/roe-dl/weewx-DWD?tab=readme-ov-file#wetterwarnungen-im-html-template) gegeben.

Für die Darstellung in der Statuszeile wie in dem Screenshot zu sehen, muß man die Erzeugung der Meldungen nachbasteln. Zunächst einmal muß in `index.html.tmp` die Bedingung mit `forecast_alert_enabled` auskommentiert werden, damit das `<div>` mit `wx-stn-alert` immer vorhanden ist. Und dann muß man `belchertown.js.tmpl` erweitern. Ich habe das in eine Datei [woellsdorf.js](https://github.com/user-attachments/files/16590716/woellsdorf.txt) ausgelagert, die ich mittels `#include` in `belchertown.js` einfüge. Natürlich müssen die Pfade innerhalb der Datei angepaßt werden. In der Funktion `update_weewx_data()` muß am Ende 

```JavaScript
ajaxdwdalert().then(show_dwd_forecast_alert);
```

eingefügt werden.

Ob die Beschreibung jetzt vollständig ist, bin ich mir nicht sicher. Es ist schon recht lange her, daß ich das gemacht habe.

**Reply by teffi71** [reply]
Herzlichen Dank für Deine schnelle Antwort und Deine Bereitschaft Deinen Code mit mir zuteilen :)

Ich habe Deine Anweisung befolgt, bekomme aber weiterhin den "normalen" Warnhinweis und nicht den DWD formatierten wie etwa [hier](http://weather.stegli.de/forecast/).

Daraufhin habe ich die Datei `woellsdorf.js` angeschaut und dies gefunden:
`async function ajaxdwd()
  {
    reply = await fetch(get_relative_url() + "/json/dwd.json");
    if (reply.ok)
      {
        return await reply.json();
      }
    else
      {
        throw new Error("HTTP error! Unable to load dwd.json");
      }
  }
`

Was ist `dwd.json`?
Bei mir finde ich keine `dwd.json`, aber im `syslog` finde ich auch keine Fehlermeldung, dass die Datei nicht gefunden wurde ...

**Reply by teffi71** [reply]
noch eine Kleinigkeit:
Du schreibst: 
> die ich mittels `#include` in `belchertown.js` einfüge

mit `belchertown.js` ist sicherlich `belchertown.js.tmpl` gemeint, oder?

**Reply by roe-dl** [reply]
Das ist unwichtig. Es wird nicht benutzt.

Entscheidend ist, daß weiter unten `"/dwd/warn-DL.json"` entsprechend angepaßt wird.

Außerdem müssen die originalen Warnmeldungen abgeschaltet werden, sonst beißt sich das gegenseitig. Deswegen der Hinweis auf `forecast_alert_enabled`.

Fehlermeldungen hierzu findet man nicht im Log des Servers sondern im Browser des Clients, bei Firefox zum Beispiel im Kontextmenü der Punkt "Untersuchen".

**Reply by teffi71** [reply]
Es scheint ein falscher Pfad zu sein ...
<img width="1294" alt="Screenshot 2024-08-13 at 16 45 14" src="https://github.com/user-attachments/assets/10c5e50b-cc13-4628-8435-2a29818f0af5">
Hmm - muss ich die Datei `warn-HI.json` auf den Web-Host hochladen?
<img width="853" alt="Screenshot 2024-08-13 at 16 52 51" src="https://github.com/user-attachments/assets/09050af7-ced9-4295-8f3a-11e000173d90">

**Reply by teffi71** [reply]
Ich hatte einen Schreibfehler in `warn-Hi.json`, denn die Datei heisst `warn-HI.json`.
Dann habe ich die Datei `warn-HI.json` händisch auf den Web-Host geladen.

Jetzt ist die obige Fehlermeldung `HTTP error! Unable to road warn-HI.json` weg, aber die DWD formatierte Warnmeldung bekomme ich immer noch nicht ...

In `index.html.tmp` habe ich die beiden Zeilen `if` und `endif` vor und nach `<div class="wx-stn-alert"><span class="wx-stn-alert-text"></span></div><!-- AJAX -->` gelöscht ...

**Comment by teffi71** [comment]
Wir haben es _fast_ geschafft 😄 
<img width="1306" alt="Screenshot 2024-08-13 at 17 53 24" src="https://github.com/user-attachments/assets/f9e834fa-4a91-43fe-abf3-b71dbe023c4d">
Es werden auch die Pop-ups richtig formatiert 😇 
<img width="1015" alt="Screenshot 2024-08-13 at 17 53 40" src="https://github.com/user-attachments/assets/4c9f8959-26ba-46cc-9a34-73e951fcb9d1">
<img width="952" alt="Screenshot 2024-08-13 at 17 53 50" src="https://github.com/user-attachments/assets/b89c1c51-a0d4-4a06-97ca-9b3b5d26cb1d">
Aber nach dem MQTT verbunden ist, werden die Warnung mit denen von AERIS überschrieben ...
<img width="821" alt="Screenshot 2024-08-13 at 17 54 06" src="https://github.com/user-attachments/assets/afc98d39-e507-43c1-9f48-05d43905dd37">
 
Jetzt schaue ich noch mal, wo ich `ajaxdwdalert().then(show_dwd_forecast_alert);` in `belchertown.js.tmpl` eingefügt habe ... 🤔

**Reply by teffi71** [reply]
`ajaxdwdalert().then(show_dwd_forecast_alert);` war bereits an letzter Stelle innerhalb der Funktion `update_weewx_data()` ... 😞

**Comment by roe-dl** [comment]
`forecast_alert_enabled` muß auf `false` gesetzt werden. Die Variable kann in `skin.conf` aber auch in `weewx.conf` stehen.

**Reply by teffi71** [reply]
Danke! Das war es! 😃 Bei mir war es in der weewx.conf `forecast_alert_enabled=1` jetzt `forecast_alert_enabled=0`.

Und ich habe noch die `warn-xyz.json` in der `skin.conf` unter `[CopyGenerator]` hinzugefügt:
```
    # List of files to be copied each time the generator runs
    copy_always = *.css,dwd/warn-HI.json
```

Nochmals herzlichen Dank für das Teilen des Codes und Deiner Unterstützung hier 🙇‍♂️
